### ISSUE TITLE
Compose tab does not work on Jolla Sailfish browser

### ISSUE THREAD

**Issue by bochecha** [mastodon/mastodon#2760]
Since last week, I can't open the compose tab on my Jolla.

(I think that corresponds to when the mstdn.fr admins upgraded to 1.3.0, but I'm not entirely sure. In any case, mstdn.fr is running Mastodon 1.3.2 now and the problem still exists)

The symptoms is that all the tabs work just fine, but when clicking on the compose tab, I get the contents of the last tab I displayed, as if the button did nothing.

The Jolla browser is based on Firefox 38, so I downloaded Firefox 38.8.0 ESR on my desktop and managed to reproduce the exact same issue by turning on the Responsive Design View and setting the viewport to 540x960, the size of my Jolla.

There, all the tabs worked just fine... except the compose tab.

When I click on the compose tab, I can see this in the Firefox console:

```
TypeError: value.keywords.includes is not a function application-de58e14cf4efdb9107448763bdfb1ffa4bbe12799d96dfe0af6d192f63434d94.js:82:3149
```

It would be really great if Mastodon worked on the Jolla browser! (of course, it would be great if Jolla updated their browser to a less ancient version of Firefox...)

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by joyeusenoelle** [comment]
Another user is reporting this as well: https://octodon.social/users/cedesguin/updates/96782

It looks like this may be an issue with the emojione-picker. I'll open an issue with that repo.

**Comment by objectifnul** [comment]
Also, I'd like to mention that webkit-based browsers on Jolla/SailfishOS (Webpirate and Webcat) don't work with any Mastodon instance.

**Comment by joyeusenoelle** [comment]
If Webkit browsers on Sailfish don't work with Mastodon at all, that's probably a separate problem; please open a new issue with any information about it you can offer.

**Comment by objectifnul** [comment]
Update 1.4.x makes the mobile web interface now completely non functional with SailfishOS :-(

**Comment by Gargron** [comment]
In our quest to reduce bundle size, we have reduced the number of polyfills in favour of expecting modern browser features to already exist. Perhaps the Jolla browser fell victim to this.

cc @nolanlawson @sorin-davidoi

**Comment by sorin-davidoi** [comment]
Latest master works find in development.
![image](https://user-images.githubusercontent.com/2109702/27771081-16de4db4-5f48-11e7-92f5-ba7749f8bd36.png)

However, loading toot.cafe results in this:
![image](https://user-images.githubusercontent.com/2109702/27771094-72bb59e2-5f48-11e7-8280-88e79b82ffe3.png)

Not really sure what is happening.

**Comment by nightpool** [comment]
@sorin-davidoi that looks like one of the polyfills is blocked by your csp? hard to tell which one it is with minimization enabled though. Can you try serving the dev code with the same CSP?

**Comment by sorin-davidoi** [comment]
> Can you try serving the dev code with the same CSP?
Not sure how to do that.

**Comment by objectifnul** [comment]
FYI, same issue with Gnu-IceCat v52.1.0 (Ubuntu 16.04) for Mastodon instances running Tootsuite 1.4x

**Comment by sorin-davidoi** [comment]
Spend some time today on this. IceCat doesn't even work in development mode. It stops screaming if I add the `Content-Security-Policy` header in `development.rb`, but still does not load the page:

```js
config.action_dispatch.default_headers = {
    'Server'                 => 'Mastodon',
    'Content-Security-Policy'=> 'script-src \'self\' http://127.0.0.1:8080 \'unsafe-eval\' \'unsafe-inline\'',
    'X-Frame-Options'        => 'DENY',
    'X-Content-Type-Options' => 'nosniff',
    'X-XSS-Protection'       => '1; mode=block',
  }
```

Don't have time to debug this further, but some useful info can be found [here](https://bauland42.com/ruby-on-rails-content-security-policy-csp/).

**Comment by nightpool** [comment]
FTR, the toot.cafe CSP is

```
Content-Security-Policy: style-src 'self' 'unsafe-inline'; script-src 'self'; 
object-src 'self'; img-src data: https:; media-src data: https:;
connect-src 'self' wss://toot.cafe; upgrade-insecure-requests
```

**Comment by nightpool** [comment]
whoops, my bad.

**Comment by nolanlawson** [comment]
toot.cafe has a pretty strict CSP but so do a lot of Mastodon instances. It appears the `base_polyfill` requires `unsafe-eval` which appears to be what we're seeing.

@bochecha Can you confirm if `base_polyfills*.js` is successfully loaded when you test?

**Comment by trwnh** [comment]
closing this issue as stale. if it can be reproduced in current versions, then it can be reopened (or a new issue can be filed)

**Comment by mbridon** [comment]
Wow, I see I had opened this issue as I had it on my Jolla 😮 

I don't have that phone anymore, and I'm using Tusky anyway, not the browser either 🤦

Sorry I never commented and thank you for closing, the issue is probably long gone anyway 🤷
